File: 3070-count-submatrices-with-top-left-element-and-sum-less-than-k/3070-count-submatrices-with-top-left-element-and-sum-less-than-k.cpp
class Solution {
public:
    int countSubmatrices(vector<vector<int>>& grid, int k) {
        int res = 0;
        vector<int> preSum(grid.size(), 0);
        for(int j = 0; j < grid[0].size(); j++)
        {
            int tot = 0;
            if(preSum[0] + grid[0][j] > k)
            {
                break;
            }
            for(int i = 0; i < grid.size(); i++)
            {
                preSum[i] += grid[i][j];
                tot += preSum[i];
                if(tot > k)
                {
                    break;
                }
                res++;
            }
        }
        return res;
    }
};

File: 3070-count-submatrices-with-top-left-element-and-sum-less-than-k/3070-count-submatrices-with-top-left-element-and-sum-less-than-k.py
class Solution:
    def countSubmatrices(self, grid: List[List[int]], k: int) -> int:
        res = 0
        for j in range(len(grid[0])):
            tot = 0
            if j > 0 and (grid[0][j - 1] + grid[0][j] > k):
                break
            for i in range(len(grid)):
                if j > 0:
                    grid[i][j] += grid[i][j - 1]
                tot += grid[i][j]
                if tot > k:
                    break
                res += 1
        return res